### PR TITLE
MRI segmentation_47640867

### PR DESCRIPTION
**Pull request by Rach1717**

### CODE REVIEW

**Comment by shaivikaaaa** [comment]
@gayanku

**Comment by wangzhaomxy** [comment]
**<This is an initial inspection, no action is required at this point.>**

**File Organizing:** Well-organized files. Please put the original README.md file back to the place under the "PatternAnalysis-2024/" folder.

**Problem Solving:**

- The algorithm solves the problem appropriately. 
- Accuracy in testing dataset: 0.9989. However, there is no evidence to support this result. 

**Model and functions:**

- It correctly uses PyTorch to construct the 2D-UNet models and functions.
- No data augmentation.
- Using original Train/Val datasets to train UNet model and observe the model performance. However, students are expected to split the train dataset into new train/val dataset to train model and observe the model performance in the training phase, then to use the original validation dataset to test the model performance on the unseen data.

**Code design:** Other parts except the custom dataset class are good.
- Poor dataset loading class: Loading the entire dataset into RAM before training.

**Code comment and docstring:**

- Minimal code comments 
- No function docstrings

**Difficulty:** Easy.

**Additional Comments:**

- Minimal and meaningless commits, just some copy-and-paste code commits.
- Scratchy report.

**Comment by gayanku** [comment]
## Marking
<table><tbody>
  <tr><td colspan='3'><b>Good/OK/Fair Practice (Design/Commenting, TF/Torch Usage)</b></td> </tr>
	<tr><td></td><td>Adequate design and implementation. </td><td>-1</td></tr>
	<tr><td></td><td>Spacing and comments. </td><td></td></tr>
	<tr><td></td><td>No Header blocks. </td><td>-1</td></tr>
	
  <tr><td colspan='3'><b>Recognition Problem</b></td></tr>
	<tr><td></td><td>OK solution to problem. No evidance of model training. Unusually high accuracy.</td><td>-2</td></tr>
	<tr><td></td><td>Driver Script present. </td><td></td></tr>
	<tr><td></td><td>File structure present. </td><td></td></tr>
	<tr><td></td><td>Good Usage & Demo & Visualisation & Data usage. </td><td></td></tr>
	<tr><td></td><td>Module present. </td><td></td></tr>
	<tr><td></td><td>Commenting missing. </td><td>-1</td></tr>
	<tr><td></td><td>No Data leakage found. </td><td></td></tr>
	<tr><td></td><td>Difficulty : Easy. UNet2D (Easy Difficulty)</td><td>-10</td></tr>
	
  <tr><td colspan='3'><b>Commit Log</b></td></tr>
	<tr><td></td><td>Some/Adequate Meaningful commit messages. </td><td>-1</td></tr>
	<tr><td></td><td>Some/Adequate Progressive commits. Some commits lack purpose.</td><td>-1</td></tr>
	
  <tr><td colspan='3'><b>Documentation</b></td></tr>
	<tr><td></td><td>Readme :Acceptable. Incomplete. No training charts/info provided.</td><td>-4</td></tr>
	<tr><td></td><td>Model/technical explanation :Missing. </td><td>-3</td></tr>
	<tr><td></td><td>Description and Comments :Good. </td><td></td></tr>
	<tr><td></td><td>Markdown used and PDF submitted. PDF checked.</td><td></td></tr>
	
  <tr><td colspan='3'><b>Pull Request</b></td></tr>
	<tr><td></td><td>Successful Pull Request (Working Algorithm Delivered on Time in Correct Branch). Submitted on time. </td><td></td></tr>
	<tr><td></td><td>No Feedback required. </td><td></td></tr>
	<tr><td></td><td>Request Description is missing. </td><td>-2</td></tr>
  <tr><td colspan='2'><b>TOTAL</b></td><td>-26</td></tr>
</tbody></table>

Marked as per the due date and changes after which aren't necessarily allowed to contribute to grade for fairness.
_Subject to approval from Shakes_
